src/components/Button/Button.tsx

```diff
@@ -16,84 +22,82 @@ interface ButtonProps {
-    mouseDown: false,
-  };
-
-  constructor(props: ButtonProps) {
-    super(props);
-  }
-
-  render(): JSX.Element {
-    const {
-      attributes,
-      blockName,
-      buttonType,
-      children,
-      className,
-      disabled,
-      id,
-      modifiers = [],
-      mouseDown,
-      onClick,
-      type = "submit",
-    } = this.props;
-
-    const buttonModifiers = modifiers ? modifiers : [];
-    if (!buttonModifiers.find((modifier) => modifier === buttonType)) {
-      buttonModifiers.push(buttonType);
-    }
-
-    const baseClass = "button";
+// Used to map between ButtonTypes enum values and Chakra variant options.
+const variantMap = {};
+for (const type in ButtonTypes) {
+  variantMap[ButtonTypes[type]] = ButtonTypes[type];
+}
 
-    const btnCallback = mouseDown
-      ? { onMouseDown: onClick }
-      : { onClick: onClick };
+/**
+ * Map the ButtonType to the Button Chakra theme variant object. If a wrong
+ * value is passed (typically in non-Typescript scenarios), then the default
+ * is the "primary" variant.
+ */
+const getVariant = (buttonType) =>
+  variantMap[buttonType] || ButtonTypes.Primary;
 
-    let childCount = 0;
-    let hasIcon = false;
+/** Renders a simple `button` element with custom classes and modifiers. */
+function Button(props: React.PropsWithChildren<ButtonProps>) {
+  const {
+    attributes,
+    buttonType,
+    children,
+    className = "",
+    disabled,
+    id,
+    mouseDown = false,
+    onClick,
+    type = "button",
+  } = props;
+  const baseClass = "button";
+  const btnCallback = mouseDown ? { onMouseDown: onClick } : { onClick };
+  let childCount = 0;
+  let hasIcon = false;
+  let variant;
+  let styles;
 
-    React.Children.map(children, (child: React.ReactElement) => {
-      childCount++;
-      if (child !== undefined && child !== null) {
-        if (
-          child.type === Icon ||
-          (child.props && child.props.mdxType === "Icon")
-        ) {
-          hasIcon = true;
-        }
+  React.Children.map(children, (child: React.ReactElement) => {
+    childCount++;
+    if (child !== undefined && child !== null) {
+      if (
+        child.type === Icon ||
+        (child.props && child.props.mdxType === "Icon")
+      ) {
+        hasIcon = true;
       }
-    });
-
-    if (childCount === 1 && hasIcon) {
-      buttonModifiers.push("icon-only");
     }
+  });
 
-    return (
-      <button
-        id={id}
-        className={`${baseClass} ${bem(baseClass, buttonModifiers, blockName, [
-          className,
-        ])}`}
-        type={type}
-        disabled={disabled}
-        {...attributes}
-        {...btnCallback}
-      >
-        {children}
-      </button>
-    );
+  if (childCount === 1 && hasIcon) {
+    variant = "icon-only";
+  } else {
+    variant = getVariant(buttonType);
   }
+
+  styles = useStyleConfig("Button", { variant });
+
+  return (
+    <ChakraButton
+      id={id}
+      className={bem(baseClass, [], "", [className])}
```

Comment:
Are we keeping this around temporarily for backwards compatibility reasons?

Reply:
Yes, and also because we haven't completely moved away from bem so maybe this should stick around until we no longer need it.

Reply:
Should we consider adding a deprecation flag comment for this and other parts that we're planning on getting rid of? 

```js
// @deprecated since version 2.3.
// Will be deleted in version 3.0.
 ```

Reply:
@nypl-wluisi Doing this on a code level seems untenable.  If we are going to flag things like this, what about adding a `Deprecated` page to the `Documentation` section in Storybook?

Reply:
Yeah, that's a good point. We can discuss more, but we need a list somewhere of all props and other things that we plan on no longer using with the switch to Chakra. And then we also need to decide if something like `bem` and `className` are things we are going to move away from, which I think we are?